src/app/qgisapp.cpp

```diff
@@ -1853,6 +1854,21 @@ bool QgisApp::event( QEvent *event )
   return done;
 }
 
+
+QgsMessageBar *QgisApp::visibleMessageBar()
```

Comment:
I quite like this. I think there's scope to use more in future -- e.g. vector/raster layer properties dialogs currently trigger message bar warnings in the main canvas, so they are a potential use case for this in future.